src/test/java/com/ait/lienzo/client/widget/panel/mediators/PanelPreviewMediatorTest.java

```diff
@@ -0,0 +1,187 @@
+        when(panel.getWidthPx()).thenReturn(WIDTH);
+        when(panel.getHeightPx()).thenReturn(HEIGHT);
+
+        previewPanel = spy(new LienzoPanelImpl(1, 1));
+
+        tested = new PanelPreviewMediator(new Supplier<LienzoBoundsPanel>() {
+            @Override
+            public LienzoBoundsPanel get() {
+                return panel;
+            }
+        }, new Consumer<IsWidget>() {
+            @Override
+            public void accept(IsWidget widget) {
+                panelContainer.add(widget);
+            }
+        }, new Supplier<LienzoPanel>() {
+            @Override
+            public LienzoPanel get() {
+                return previewPanel;
+            }
+        });
+        tested.setMaxScale(MAX_SCALE);
+    }
+
+    @Test
+    public void testConstruction() {
+        verify(panelContainer, times(1)).add(any(IsWidget.class));
+        assertNotNull(tested.getPreviewLayer());
+        PreviewLayer previewLayer = tested.getPreviewLayer();
+        assertTrue(previewLayer.isListening());
+        assertTrue(previewLayer.isTransformable());
+        assertNotNull(tested.getPreviewPanel());
+        LienzoPanel previewPanel = tested.getPreviewPanel();
+        assertFalse(previewPanel.isVisible());
+        assertEquals(previewLayer, previewLayer.getLayer());
+        assertNotNull(tested.getMediator());
+        MouseBoxZoomMediator mediator = tested.getMediator();
+        assertFalse(mediator.isEnabled());
+        assertEquals(mediator, previewLayer.getViewport().getMediators().pop());
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testEnable() {
+        Element previewPanelElement = mock(Element.class);
+        Style previewPanelStyle = mock(Style.class);
+        when(previewPanel.getElement()).thenReturn(previewPanelElement);
+        when(previewPanelElement.getStyle()).thenReturn(previewPanelStyle);
+        when(panel.getLayerBounds()).thenReturn(Bounds.build(0, 0, WIDTH / 2, HEIGHT / 2));
+        tested.enable();
+        assertTrue(tested.getPreviewLayer().isListening());
+        assertFalse(layer.isListening());
+        assertFalse(layer.isVisible());
+        verify(previewPanelStyle, times(1)).setPosition(eq(Style.Position.ABSOLUTE));
+        verify(previewPanelStyle, times(1)).setTop(eq(0d), eq(Style.Unit.PX));
+        verify(previewPanelStyle, times(1)).setLeft(eq(0d), eq(Style.Unit.PX));
+        verify(previewPanelStyle, times(1)).setBorderStyle(eq(Style.BorderStyle.NONE));
+        verify(previewPanelStyle, times(1)).setBackgroundColor(eq(PanelPreviewMediator.PREVIEW_BG_COLOR));
+        verify(previewPanel, times(1)).setPixelSize(WIDTH, HEIGHT);
+        verify(previewPanel, times(1)).setVisible(eq(false));
+        Transform previewTransform = tested.getPreviewLayer().getViewport().getTransform();
+        assertEquals(0.5833333333333334d, previewTransform.getScaleX(), 0d);
+        assertEquals(0.5833333333333334d, previewTransform.getScaleY(), 0d);
```

Comment:
I see in the init we set MaxScale to 0.5 and here it is bigger, could you please explain little bit how this number is computed?

Reply:
sure, that's because this number is not the real scale factor that the preview mediator will apply, which as you said cannot be bigger than 0.5. 
This number is the scale factor being applied only the preview mediator is being enabled, because once enabled the whole canvas is being scaled down in order to fit the actual panel size. At this point, the user can draw a rectangle to select an area to jump into, so once drawing the ractangle and doing the mouse up, it applies the scale factor that will result the visible area selected being used as for the panel current size. This one is the real scale factor, which cannot be bigger than 0.5.